src/receipt.rs

```diff
@@ -183,7 +187,24 @@ impl Receipt {
     }
 
     pub const fn panic(id: ContractId, reason: InstructionResult, pc: Word, is: Word) -> Self {
-        Self::Panic { id, reason, pc, is }
+        Panic {
+            id,
+            reason,
+            pc,
+            is,
+            contract_id: None,
+        }
+    }
+
+    pub fn set_panic_contract_id(mut self, _contract_id: Option<ContractId>) -> Self {
```

Comment:
Maybe a (slightly) nicer name would be `with_` instead of `set_` since you are consuming and returning it, instead of setting it.